src/Caching/Hybrid/src/Runtime/HybridCache.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Distributed;
+
+namespace Microsoft.Extensions.Caching.Hybrid;
+
+/// <summary>
+/// Provides multi-tier caching services building on <see cref="IDistributedCache"/> backends.
+/// </summary>
+public abstract class HybridCache
+{
+    /// <summary>
+    /// Get data from the cache, or the underlying data service if not available.
+    /// </summary>
+    /// <typeparam name="T">The type of the data being considered.</typeparam>
+    /// <typeparam name="TState">The type of additional state required by <paramref name="underlyingDataCallback"/>.</typeparam>
+    /// <param name="key">The unique key for this cache entry.</param>
+    /// <param name="underlyingDataCallback">Provides the underlying data service is the data is not available in the cache.</param>
+    /// <param name="state">Additional state required for <paramref name="underlyingDataCallback"/>.</param>
+    /// <param name="options">Additional options for this cache entry.</param>
+    /// <param name="tags">The tags to associate with this cache item.</param>
+    /// <param name="token">Cancellation for this operation.</param>
+    /// <returns>The data, either from cache or the underlying data service.</returns>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Delegate differences make this unambiguous")]
+    public abstract ValueTask<T> GetOrCreateAsync<TState, T>(string key, TState state, Func<TState, CancellationToken, ValueTask<T>> underlyingDataCallback,
```

Comment:
```suggestion
    public abstract ValueTask<T> GetOrCreateAsync<TState, T>(string key, TState state, Func<TState, CancellationToken, ValueTask<T>> factory,
```
Should the name be `factory` instead? It seems clearer and follows the same convention as in [`Microsoft.Extensions.Caching.Abstractions`](https://github.com/dotnet/runtime/blob/b2db8513a9c759587efea5cc00e28a2241c8572c/src/libraries/Microsoft.Extensions.Caching.Abstractions/src/MemoryCacheExtensions.cs#L208).